Improvements.rst

```diff
@@ -0,0 +1,447 @@
++----------------------------------------------------------------+------+------+------+-------+------+-------+-------+-------+-------+--------+
+| Another argument stack optimization (psr -> dr call)           | 2818 | 3281 | 1934 | 8033  | 2217 | 21.6% | 64017 | 66370 | 37220 | 167607 |
++----------------------------------------------------------------+------+------+------+-------+------+-------+-------+-------+-------+--------+
+| ^^^ Finally this solution is better then original in **EVERY** test suite branch! The last hurdle - Exter GGC is now less!!! ^^^            |
++----------------------------------------------------------------+------+------+------+-------+------+-------+-------+-------+-------+--------+
+| Black magic route optimization (drop some result later)        | 2818 | 3281 | 1916 | 8015  | 2235 | 21.8% | 64017 | 66370 | 37130 | 167517 |
++----------------------------------------------------------------+------+------+------+-------+------+-------+-------+-------+-------+--------+
+| Another black magic optimization (drop auth_kind later)        | 2818 | 3281 | 1906 | 8005  | 2245 | 21.9% | 64017 | 66370 | 37102 | 167489 |
++----------------------------------------------------------------+------+------+------+-------+------+-------+-------+-------+-------+--------+
+| Backported FunC optimizations from entrypoint branch           | 2782 | 3373 | 1824 | 7979  | 2245 | 22.1% | 60011 | 64810 | 34138 | 158959 |
++----------------------------------------------------------------+------+------+------+-------+------+-------+-------+-------+-------+--------+
+| Use SDBEGINS to enforce prefix in external message             | 2710 | 3373 | 1824 | 7907  | 2343 | 22.9% | 59147 | 64810 | 34138 | 158095 |
++----------------------------------------------------------------+------+------+------+-------+------+-------+-------+-------+-------+--------+
+| Use SDBEGINSQ to check internal message prefixes               | 2710 | 3283 | 1736 | 7729  | 2521 | 24.6% | 59237 | 64090 | 33578 | 156905 |
++----------------------------------------------------------------+------+------+------+-------+------+-------+-------+-------+-------+--------+
+| Backport some optimizations from EP and coalesce code          | 2699 | 3165 | 1828 | 7692  | 2558 | 25.0% | 59108 | 63031 | 34141 | 156280 |
++----------------------------------------------------------------+------+------+------+-------+------+-------+-------+-------+-------+--------+
+| Optimized instructions order for extension and fix args        | 2699 | 3165 | 1810 | 7674  | 2576 | 25.1% | 59108 | 63031 | 34033 | 156172 |
++----------------------------------------------------------------+------+------+------+-------+------+-------+-------+-------+-------+--------+
+| *Reminder and origin point: INITIAL*                           | 3235 | 4210 | 2760 | 10250 | 0    | 0.00% | 64038 | 71163 | 38866 | 174067 |

```

Comment:
Wrong initial total, should be 10205 instead of 10250

Reply:
Well, that's a difficult one, I see, but that will require recalculating of all the percentages and differences. Will work on it.

Reply:
Fixed and recalculated everything